definitions/tools/mutect.cwl

```diff
@@ -2,75 +2,62 @@
 
 cwlVersion: v1.0
 class: CommandLineTool
-label: "mutect2 (GATK 3.6)"
-baseCommand: ["/usr/bin/java", "-jar", "/opt/GenomeAnalysisTK.jar", "-T", "MuTect2"]
+label: "Mutect2 (GATK 4)"
+
+baseCommand: ["/bin/bash", "Mutect2.sh"]
+
 requirements:
     - class: ResourceRequirement
       ramMin: 20000
       tmpdirMin: 100000
     - class: DockerRequirement
-      dockerPull: "mgibio/cle:v1.3.1"
+      dockerPull: "broadinstitute/gatk:4.1.2.0"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'Mutect2.sh'
+        entry: |
+            set -o pipefail
+            set -o errexit
+            
+            TUMOR=`(samtools view -H $3 | grep "^@RG" | sed "s/.*SM:\([^\t]*\).*/\1/g" | head -n 1)` #Extracting the sample name from the TUMOR bam.
+            NORMAL=`(samtools view -H $4 | grep "^@RG" | sed "s/.*SM:\([^\t]*\).*/\1/g" | head -n 1)` #Extracting the sample name from the NORMAL bam.
+            /gatk/gatk Mutect2 -O $1 -R $2 -I $3 -tumor $TUMOR -I $4 -normal $NORMAL -L $5 #Running Mutect2.
+            gunzip mutect.vcf.gz #Unzipping the Mutect2 output vcf file to alter the header names.
+            grep "^##" mutect.vcf >mutect.final.vcf #Extracting all the lines above #CHROM in the vcf to a new output vcf.
+            grep "^#CHROM" mutect.vcf | sed "s/\t$TUMOR\t/\tTUMOR\t/g; s/\t$NORMAL/\tNORMAL/g" >> mutect.final.vcf #Concatenating the #CHROM line with substituted sample headers to the output vcf.
```

Comment:
This is kind of an odd thing to have to do.  Is this because the new version of Mutect does something different from the old one?

Reply:
The new version of Mutect requires the VCF files to have the the sample names as the headers, cause it matches it from the BAM file.